olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -309,30 +284,63 @@ bool DefaultCacheImpl::RemoveKeysWithPrefix(const std::string& key) {
   RemoveKeysWithPrefixLru(key);
 
   if (mutable_cache_) {
-    return mutable_cache_->RemoveKeysWithPrefix(key);
+    uint64_t removed_data_size = 0;
+    auto result = mutable_cache_->RemoveKeysWithPrefix(key, removed_data_size);
+    mutable_cache_data_size_ -= removed_data_size;
+    return result;
   }
   return true;
 }
 
 void DefaultCacheImpl::InitializeLru() {
-  if (!mutable_cache_ || settings_.max_disk_storage == kMaxDiskSize ||
-      settings_.eviction_policy == EvictionPolicy::kNone) {
+  if (!mutable_cache_) {
     return;
   }
 
-  mutable_cache_lru_ =
-      std::make_unique<DiskLruCache>(settings_.max_disk_storage);
+  if (mutable_cache_ && settings_.max_disk_storage != kMaxDiskSize &&
+      settings_.eviction_policy == EvictionPolicy::kLeastRecentlyUsed) {
+    mutable_cache_lru_ =
+        std::make_unique<DiskLruCache>(settings_.max_disk_storage);
+  }
+
+  if (mutable_cache_lru_) {
+    OLP_SDK_LOG_INFO_F(kLogTag, "Initializing mutable lru cache.");
+  }
+
+  const auto start = std::chrono::steady_clock::now();
+  auto count = 0u;
   auto it = mutable_cache_->NewIterator(leveldb::ReadOptions());
   for (it->SeekToFirst(); it->Valid(); it->Next()) {
     const auto key = it->key().ToString();
     const auto& value = it->value();
+    mutable_cache_data_size_ += key.size() + value.size();
 
-    // Skip expiry keys
-    if (IsExpiryKey(key)) {
-      continue;
+    if (mutable_cache_lru_) {
+      // Skip expiry keys
+      if (IsExpiryKey(key)) {
+        continue;
+      }
```

Comment:
For next PR!
Can be merged to:
```suggestion
    if (mutable_cache_lru_ && !IsExpiryKey(key)) {
     ++count
     ....
     }
```